src/transfers/live.rs

```diff
@@ -93,45 +84,33 @@ impl<T: DuplexTransport + 'static> Future for WatchLiveLogs<T> {
     type Item = ();
     type Error = web3::Error;
     fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
+        let handle = self.handle.clone();
         loop {
-            let handle = self.handle.clone();
-            let next = match self.state {
-                SubscriptionState::Subscribing(ref mut future) => {
-                    let stream = try_ready!(future.poll());
-                    Some(SubscriptionState::Subscribed(stream))
+            match self.stream.poll() {
+                Ok(Async::Ready(Some(log))) => {
+                    // On receiving a valid log, spawn the waiting for a receipt, as it takes
+                    // 20 blocks which is 20 seconds or 300 seconds and we don't want to block
+                    handle.spawn(self.process_log(&log));
```

Comment:
Is this guaranteed to return? What happens if it times out?

Reply:
Process log can return an error. If it does, the spawned task is killed, and it logs a message about not processing that log. 
Everything else, including the WatchLiveLogs future continues working. 
If it was a transfer, it should get grabbed and approved by the past transfer subscription. 
If it was a flush, approvals are blocked since they read off the chain, and the relay won't actually withdraw funds until restarted.